driver.go

```diff
@@ -112,6 +112,9 @@ func (d MySQLDriver) Open(dsn string) (driver.Conn, error) {
 		mc.cleanup()
 		return nil, err
 	}
+	if plugin == "" {
+		plugin = defaultAuthPlugin
+	}
```

Comment:
See https://github.com/PyMySQL/PyMySQL/pull/709#discussion_r200921337

When server sent "" as plugin name, it's safe to response "" as plugin_name too.